test/extensions/filters/http/ext_proc/ext_proc_benchmark_test.cc

```diff
@@ -0,0 +1,254 @@
+      // Create a cluster for our gRPC server pointing to the address that is running the gRPC
+      // server.
+      auto* processor_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      processor_cluster->set_name("ext_proc_server");
+      processor_cluster->mutable_load_assignment()->set_cluster_name("ext_proc_server");
+      auto* address = processor_cluster->mutable_load_assignment()
+                          ->add_endpoints()
+                          ->add_lb_endpoints()
+                          ->mutable_endpoint()
+                          ->mutable_address()
+                          ->mutable_socket_address();
+      address->set_address(Network::Test::getLoopbackAddressString(ipVersion()));
+      address->set_port_value(test_processor_.port());
+
+      // Ensure "HTTP2 with no prior knowledge." Necessary for gRPC.
+      ConfigHelper::setHttp2(
+          *(bootstrap.mutable_static_resources()->mutable_clusters()->Mutable(0)));
+      ConfigHelper::setHttp2(*processor_cluster);
+
+      // Make sure both flavors of gRPC client use the right address.
+      const auto addr = Network::Test::getCanonicalLoopbackAddress(ipVersion());
+      const auto addr_port = Network::Utility::getAddressWithPort(*addr, test_processor_.port());
+      setGrpcService(*proto_config_.mutable_grpc_service(), "ext_proc_server", addr_port);
+
+      // Merge the filter.
+      envoy::config::listener::v3::Filter ext_proc_filter;
+      ext_proc_filter.set_name("envoy.filters.http.ext_proc");
+      ext_proc_filter.mutable_typed_config()->PackFrom(proto_config_);
+      config_helper_.prependFilter(MessageUtil::getJsonStringFromMessageOrDie(ext_proc_filter));
+    });
+
+    setUpstreamProtocol(Http::CodecType::HTTP2);
+    setDownstreamProtocol(Http::CodecType::HTTP2);
+    HttpIntegrationTest::initialize();
+  }
+
+  int getTestIterations() const {
+    const auto env_value = TestEnvironment::getOptionalEnvVar("EXT_PROC_BENCHMARK_ITERATIONS");
+    int iterations;
+    if (env_value && absl::SimpleAtoi(*env_value, &iterations)) {
+      return iterations;
+    }
+    return DefaultTestIterations;
+  }
+
+  int testIterations() {
+    static int iterations = getTestIterations();
+    return iterations;
+  }
+
+  void measureHttpGets(absl::string_view test_name, int response_size = 100) {
+    // The PERF_RECORD macro is weird and we need to use this variable for something
+    // else or we will fail to compile.
+    EXPECT_FALSE(test_name.empty());
+    for (int iteration = 0; iteration < getTestIterations(); iteration++) {
+      Http::TestRequestHeaderMapImpl headers;
+      HttpTestUtility::addDefaultHeaders(headers);
+      headers.addCopy(Http::LowerCaseString("response_size_bytes"), response_size);
+      auto conn = makeClientConnection(lookupPort("http"));
+      codec_client_ = makeHttpConnection(std::move(conn));
+
+      PERF_OPERATION(op);
```

Comment:
I think it would be good to add perf annotations to the ext_proc filter itself to get better visibility into the processing time without I/O.